plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ServiceResourceMapper.java

```diff
@@ -0,0 +1,163 @@
+package org.jboss.tools.openshift.internal.ui.models;
+
+import static org.jboss.tools.openshift.internal.core.util.ResourceUtils.containsAll;
+import static org.jboss.tools.openshift.internal.core.util.ResourceUtils.imageRef;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.stream.Collectors;
+
+import org.jboss.tools.openshift.core.OpenShiftAPIAnnotations;
+
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IBuild;
+import com.openshift.restclient.model.IBuildConfig;
+import com.openshift.restclient.model.IDeploymentConfig;
+import com.openshift.restclient.model.IPod;
+import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.IService;
+import com.openshift.restclient.model.deploy.DeploymentTriggerType;
+import com.openshift.restclient.model.deploy.IDeploymentImageChangeTrigger;
+import com.openshift.restclient.model.route.IRoute;
+
+/**
+ * This class encapsulates the logic used for determinining which resource in a
+ * project are related to whilch service.
+ * 
+ * @author thomas
+ *
+ */
+public class ServiceResourceMapper {
+	public static final Collection<IResource> computeRelatedResources(IService s, Collection<IResource> resources) {
+		Collection<IResource> result = new HashSet<>();
+
+		resources.forEach(resource -> {
+			if (resource instanceof IPod) {
+				if (isRelated(s, (IPod) resource)) {
+					result.add(resource);
+					result.addAll(getRelated(resources, (IPod) resource));
+				}
+			} else if (resource instanceof IDeploymentConfig) {
+				if (isRelated(s, (IDeploymentConfig) resource)) {
+					result.add(resource);
+					result.addAll(getRelated(resources, (IDeploymentConfig) resource));
+				}
+			} else if (resource instanceof IRoute) {
+				if (isRelated(s, (IRoute)resource)) {
+					result.add(resource);
+				}
+			}
+		});
+		return result;
+	}
+
+	private static boolean isRelated(IService s, IRoute resource) {
+		return resource.getServiceName().equals(s.getName());

```

Comment:
This method reimplements ResourceUtils.areRelated(IRoute route, IService service) without checks. If this implementation is safe and correct, should not it replace the method in ResourceUtils? Other 'isRelated' methods could go to ResourceUtils as well.